pkg/ddc/alluxio/worker.go

```diff
@@ -32,57 +27,31 @@ import (
 // over the status by setting phases and conditions. The function
 // calls for a status update and finally returns error if anything unexpected happens.
 func (e *AlluxioEngine) SetupWorkers() (err error) {
-	runtime, err := e.getRuntime()
-	if err != nil {
-		e.Log.Error(err, "setupWorker")
-		return err
-	}
-
-	replicas := runtime.Replicas()
+	var (
+		workerName string = e.getWorkertName()
+		namespace  string = e.namespace
+	)
 
-	currentReplicas, err := e.AssignNodesToCache(replicas)
+	workers, err := kubeclient.GetStatefulSet(e.Client, workerName, namespace)
 	if err != nil {
 		return err
 	}
 
-	e.Log.Info("check the desired and current replicas",
-		"desiredReplicas", replicas,
-		"currentReplicas", currentReplicas)
-
-	if currentReplicas == 0 {
-		return fmt.Errorf("the number of the current workers which can be scheduled is 0")
-	}
-
-	// 2. Update the status
 	err = retry.RetryOnConflict(retry.DefaultBackoff, func() error {
 		runtime, err := e.getRuntime()
 		if err != nil {
-			e.Log.Error(err, "setupWorker")
 			return err
 		}
-
 		runtimeToUpdate := runtime.DeepCopy()
-
-		runtimeToUpdate.Status.WorkerPhase = datav1alpha1.RuntimePhaseNotReady
-		runtimeToUpdate.Status.DesiredWorkerNumberScheduled = replicas
-		runtimeToUpdate.Status.CurrentWorkerNumberScheduled = currentReplicas
-
-		if len(runtimeToUpdate.Status.Conditions) == 0 {
-			runtimeToUpdate.Status.Conditions = []datav1alpha1.RuntimeCondition{}
-		}
-		cond := utils.NewRuntimeCondition(datav1alpha1.RuntimeWorkersInitialized, datav1alpha1.RuntimeWorkersInitializedReason,
-			"The workers are initialized.", corev1.ConditionTrue)
-		runtimeToUpdate.Status.Conditions =
-			utils.UpdateRuntimeCondition(runtimeToUpdate.Status.Conditions,
-				cond)
-
-		if !reflect.DeepEqual(runtime.Status, runtimeToUpdate.Status) {
-			return e.Client.Status().Update(context.TODO(), runtimeToUpdate)
+		err = e.Helper.SetupWorkers(runtimeToUpdate, runtimeToUpdate.Status, workers)
```

Comment:
One suggestion:
Return only enough error messages
Print the error message to the top level for processing
Error messages like this can not help us locate the problem, especially if we have multiple instances
e.g.: 
```go
return errors.Wrapf(err,"failed to set alluxio runtime worker name:%s,namespace:%s", 
                                  runtimeToUpdate.Name, 
                                  runtimeToUpdate.Namespace)
```

Reply:
Good suggestion! I will add enough error messages to help locate the problem